logging/logfile.go

```diff
@@ -93,23 +93,27 @@ func (l *LogFile) pruneFiles() error {
 	if l.MaxFiles == 0 {
 		return nil
 	}
-	pattern := l.fileNamePattern()
-	//get all the files that match the log file pattern
-	globExpression := filepath.Join(l.logPath, fmt.Sprintf(pattern, "*"))
-	matches, err := filepath.Glob(globExpression)
+
+	pattern := filepath.Join(l.logPath, fmt.Sprintf(l.fileNamePattern(), "*"))
+	matches, err := filepath.Glob(pattern)
 	if err != nil {
 		return err
 	}
-	var stale int
-	if l.MaxFiles <= -1 {
-		// Prune everything
-		stale = len(matches)
-	} else {
-		// Prune if there are more files stored than the configured max
-		stale = len(matches) - l.MaxFiles
+
+	switch {
```

Comment:
Nit: A comment above this statement would be awesome, not necessary since it isn't that complicated, but would be a nice touch.